File: plugins/kotlin/j2k/new/tests/testData/newJ2k/javaStandardMethods/cloneable.java
class C1 {
    public Object clone() throws CloneNotSupportedException {
        return super.clone();
    }
}

class C2 implements Cloneable {
    public Object clone() throws CloneNotSupportedException {
        return super.clone();
    }
}

class C3 extends C2 {
    public Object clone() throws CloneNotSupportedException {
        return super.clone();
    }
}

class C4 extends C2 implements Cloneable {
    public Object clone() throws CloneNotSupportedException {
        return super.clone();
    }
}

File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/javaStandardMethods/cloneable.kt
internal class C1 : Cloneable {
    @Throws(CloneNotSupportedException::class)
    public override fun clone(): Any {
        return super.clone()
    }
}

internal open class C2 : Cloneable {
    @Throws(CloneNotSupportedException::class)
    public override fun clone(): Any {
        return super.clone()
    }
}

internal class C3 : C2() {
    @Throws(CloneNotSupportedException::class)
    override fun clone(): Any {
        return super.clone()
    }
}

internal class C4 : C2() {
    @Throws(CloneNotSupportedException::class)
    override fun clone(): Any {
        return super.clone()
    }
}
